src/test/java/teammates/test/cases/action/StudentCourseJoinAuthenticatedActionTest.java

```diff
@@ -279,4 +282,50 @@ public void testExecuteAndPostProcess() throws Exception {
     protected StudentCourseJoinAuthenticatedAction getAction(String... params) {
         return (StudentCourseJoinAuthenticatedAction) gaeSimulation.getActionObject(getActionUri(), params);
     }
+
+    @Override
+    @Test
+    protected void testAccessControl() throws Exception {
+        dataBundle = getTypicalDataBundle();
+        removeAndRestoreDataBundle(dataBundle);
```

Comment:
Could make some changes to `StudentCourseJoinAuthenticatedTest.json` to include an unregistered user, rather than loading another data bundle.